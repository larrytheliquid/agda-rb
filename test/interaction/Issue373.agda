module Issue373 where

data ⊤ : Set where
    tt : ⊤

{-# COMPILED_DATA ⊤ () () #-}

data ℕ : Set where
  zero : ℕ
  suc  : (n : ℕ) → ℕ

{-# BUILTIN NATURAL ℕ    #-}
{-# BUILTIN ZERO    zero #-}
{-# BUILTIN SUC     suc  #-}
{-# IMPORT Imports.Nat #-}
{-# COMPILED_DATA ℕ Imports.Nat.Nat Imports.Nat.Zero Imports.Nat.Suc #-}

data List (A : Set) : Set where
  []   : List A
  _∷_ : A → List A → List A

{-# BUILTIN LIST List #-}
{-# BUILTIN NIL  []   #-}
{-# BUILTIN CONS _∷_  #-}
{-# COMPILED_DATA List [] [] (:) #-}

postulate
  String : Set

{-# BUILTIN STRING String #-}
{-# COMPILED_TYPE String String #-}

postulate
  IO : Set → Set

{-# BUILTIN IO IO #-}
{-# COMPILED_TYPE IO IO #-}

infixl 1 _>>=_

postulate
  _>>=_ : ∀ {A B} → IO A → (A → IO B) → IO B

{-# COMPILED _>>=_ (\_ _ -> (>>=) :: IO a -> (a -> IO b) -> IO b) #-}

postulate
  putStrLn : String → IO ⊤

{-# COMPILED putStrLn putStrLn #-}

f : ℕ → String
f zero = "bad"
f _    = "ok"

-- Works:

-- main = putStrLn (f (suc zero))

-- Compiles, but when the program is run we (used to) get the output
-- "bad":

main = putStrLn (f 1)
