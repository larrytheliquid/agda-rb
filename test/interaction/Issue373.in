import System.FilePath
import System.Process

let objectFile = "Imports/Nat.o"
ex <- System.Directory.doesFileExist objectFile
if ex then System.Directory.removeFile objectFile else return ()

top_command (cmd_compile MAlonzo currentFile [])

let binary = "." </> dropExtension currentFile
runProcess binary [] Nothing Nothing Nothing Nothing Nothing
