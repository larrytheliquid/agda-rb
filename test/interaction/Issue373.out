agda2_mode_code (agda2-status-action "")
[1 of 3] Compiling MAlonzo.RTE      ( MAlonzo/RTE.hs, MAlonzo/RTE.o )
[2 of 3] Compiling Imports.Nat      ( Imports/Nat.hs, Imports/Nat.o )
[3 of 3] Compiling MAlonzo.Code.Issue373 ( MAlonzo/Code/Issue373.hs, MAlonzo/Code/Issue373.o )
agda2_mode_code (agda2-status-action "Checked")
agda2_mode_code (agda2-info-action
                 "*Compilation result*"
                 "The module was successfully compiled.")
agda2_mode_code (agda2-goals-action '())
ok
